--- jdapimin.c	2012-04-04 01:09:00.000000000 +0800
+++ jdapimin-patched.c	2012-04-04 01:42:44.000000000 +0800
@@ -36,9 +36,9 @@
   cinfo->mem = NULL;		/* so jpeg_destroy knows mem mgr not called */
   if (version != JPEG_LIB_VERSION)
     ERREXIT2(cinfo, JERR_BAD_LIB_VERSION, JPEG_LIB_VERSION, version);
-  if (structsize != SIZEOF(struct jpeg_decompress_struct))
-    ERREXIT2(cinfo, JERR_BAD_STRUCT_SIZE, 
-	     (int) SIZEOF(struct jpeg_decompress_struct), (int) structsize);
+  /*if (structsize != SIZEOF(struct jpeg_decompress_struct))*/
+    /*ERREXIT2(cinfo, JERR_BAD_STRUCT_SIZE, */
+		 /*(int) SIZEOF(struct jpeg_decompress_struct), (int) structsize);*/
 
   /* For debugging purposes, we zero the whole master structure.
    * But the application has already set the err pointer, and may have set
