--- lvrend.cpp	2012-04-24 10:27:33.000000000 +0800
+++ lvrend-patched.cpp	2012-04-24 10:27:28.000000000 +0800
@@ -1902,7 +1902,8 @@
     UPDATE_STYLE_FIELD( font_style, css_fs_inherit );
     UPDATE_STYLE_FIELD( font_weight, css_fw_inherit );
     UPDATE_STYLE_FIELD( font_family, css_ff_inherit );
-    UPDATE_STYLE_FIELD( font_name, "" );
+    //UPDATE_STYLE_FIELD( font_name, "" );
+	pstyle->font_name = parent_font.get()->getTypeFace();
     UPDATE_LEN_FIELD( font_size );
     //UPDATE_LEN_FIELD( text_indent );
     spreadParent( pstyle->text_indent, parent_style->text_indent );
